src/snmalloc/mem/sizeclasstable.h

```diff
@@ -137,7 +137,9 @@ namespace snmalloc
     // Client responsible for checking alignment is a power of two
     SNMALLOC_ASSERT(bits::is_pow2(alignment));
 
-    return ((alignment - 1) | (size - 1)) + 1;
+    size_t result = ((alignment - 1) | (size - 1)) + 1;
+    // Check for overflow and return the maximum size.
+    return result < size ? result - 1 : result;
```

Comment:
I _think_ (and I think z3 agrees, see below) that `result < size` implies `result == 0`, but it'd be good to write that down or perhaps use the other spelling?

```
(define-fun result ( (size (_ BitVec 32)) (align (_ BitVec 32)) ) (_ BitVec 32)
  (bvadd #x00000001 (bvor (bvsub align #x00000001) (bvsub size #x00000001))))

(declare-const align (_ BitVec 32))
(declare-const size (_ BitVec 32))

(define-fun conjecture () Bool
  (=> (bvult (result size align) size) (= (result size align) #x00000000)))
(assert (not conjecture))
(check-sat)
```

Reply:
Agreed.  I used result-1 as it was a quick way to give -1 of the right size.  I have moved to SIZE_MAX now, and also I rewrote it to generate better code with the sizeclass calc.